### PR TITLE
Refactor file, cache, xmldata, task and regex BBEs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/directory-listener/directory_listener.bal [hunk 1]
```diff
@@ -15,19 +15,19 @@ listener file:Listener inFolder = new ({
 // The directory listener should have at least one of these predefined resources.
 service "localObserver" on inFolder {
 
-    // This resource is invoked once a new file is created in the listening directory.
+    // This function is invoked once a new file is created in the listening directory.
     remote function onCreate(file:FileEvent m) {
         string msg = "Create: " + m.name;
         log:printInfo(msg);
```
Comment (daneshk) [inline examples/directory-listener/directory_listener.bal:21]: can merge these two lines

File: examples/cache/cache.description [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // The Ballerina Cache API by-default provides an in-memory cache implementation with an
```
Comment (praneesha) [inline examples/cache/cache.description:1]:
```suggestion
// The Ballerina Cache API provides an in-memory cache implementation by default  with a
```

File: examples/cache/cache.description [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
-// LRU algorithm based eviction policy. The caching implementation can be customized by providing different
+// Least Recently Used algorithm based eviction policy. The caching implementation can be customized by providing different
```
Comment (praneesha) [inline examples/cache/cache.description:2]:
```suggestion
// `Least Recently Used` algorithm-based eviction policy. The caching implementation can be customized by providing different
```

File: examples/cache/cache.metatags [hunk 4]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to perform an in-memory caching with the Least Recently Used algorithm in Ballerina.
```
Comment (praneesha) [inline examples/cache/cache.metatags:1]:
```suggestion
description: This BBE shows how to perform an in-memory caching with the "Least Recently Used" algorithm in Ballerina.
```

File: examples/file/file.bal [hunk 5]
```diff
@@ -34,44 +25,32 @@ public function main() returns error? {
...
-    // Remove the directory in the specified file path with all its children.
-    removeResults = file:remove("foo", file:RECURSIVE);
-    if (removeResults is ()) {
-        io:println("Removed 'foo' directory with all the children.");
-    }
+    // Removes the directory in the specified file path with all its children.
+    check file:remove("foo", file:RECURSIVE);
 
-    // Create a temporary file in the default tmp directory of the OS
+    // Creates a temporary file in the default tmp directory of the OS
```
Comment (praneesha) [inline examples/file/file.bal:49]:
```suggestion
    // Creates a temporary file in the default `tmp` directory of the OS.
```

File: examples/file/file.bal [hunk 6]
```diff
@@ -34,44 +25,32 @@ public function main() returns error? {
...
     string tmpResult = check file:createTemp();
     io:println("Absolute path of the tmp file: ", tmpResult);
 
-    // Create a temporary file in a specific directory
+    // Creates a temporary file in a specific directory
```
Comment (praneesha) [inline examples/file/file.bal:53]:
```suggestion
    // Creates a temporary file in a specific directory.
```

File: examples/filepath/file.metatags [hunk 7]
```diff
@@ -0,0 +1,2 @@
+description: This BBE shows how to manipulate file paths in a way that is compatible with the target operating system in Ballerina.
```
Comment (praneesha) [inline examples/filepath/file.metatags:1]:
```suggestion
description: This BBE shows how to manipulate file paths in a way that is compatible with the target Operating System in Ballerina.
```

File: examples/task-frequency-job-execution/task_frequency_job_execution.bal [hunk 8]
```diff
@@ -22,20 +22,20 @@ class Job {
...
+    // Schedules the task to execute the job every second.
     task:JobId id = check task:scheduleJobRecurByFrequency(new Job(0),
                                         1, startTime = time);
 
-    // Wait for twelve seconds.
+    // Waits for twelve seconds.
     runtime:sleep(12);
 
-    // UnSchedule the job.
+    // UnSchedules the job.
```
Comment (praneesha) [inline examples/task-frequency-job-execution/task_frequency_job_execution.bal:39]:
```suggestion
    // Unschedules the job.
```

File: examples/task-one-time-job-execution/task_one_time_job_execution.bal [hunk 9]
```diff
@@ -9,7 +9,7 @@ class Job {
     *task:Job;
     int i = 1;
 
-    // Get executed by the scheduler when the scheduled trigger fires.
+    // Executes by the scheduler when the scheduled trigger fires.
```
Comment (praneesha) [inline examples/task-one-time-job-execution/task_one_time_job_execution.bal:12]:
```suggestion
    // Executes the scheduler when the scheduled trigger fires.
```

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

**Review by daneshk — APPROVED** [review]
LGTM

**Review by praneesha — APPROVED** [review]

**Pull request merged**